### PR TITLE
configuration properties for frame size and control stream dispatcher

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -171,15 +167,40 @@ akka {
...
         # Settings for the materializer that is used for the remote streams.
         materializer = ${akka.stream.materializer}
-        materializer {
-          dispatcher = "akka.remote.default-remote-dispatcher"
-        }
-
+        
+        # If set to a nonempty string artery will use the given dispatcher for
+        # the ordinary and large message streams, otherwise the default dispatcher is used.
+        use-dispatcher = "akka.remote.default-remote-dispatcher"
```
Comment (johanandren) [inline akka-remote/src/main/resources/reference.conf]: Why the `use`-prefix?

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-remote/src/main/resources/reference.conf (hunk 1 above)
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: it's a convention we have at other places, when it's a dispatcher name, i.e. a reference to a dispatcher configuration section (as opposed to the dispatcher section itself)

Comment (johanandren) [inline akka-remote/src/main/resources/reference.conf]: Ok, thanks for clarifying!

File: akka-remote/src/main/resources/reference.conf [hunk 2]
```diff
@@ -171,15 +167,40 @@ akka {
       log-sent-messages = off
 
       advanced {
+      
+        # Maximum serialized message size, including header data.  
+        maximum-frame-size = 256 KiB
+        
+        # Direct byte buffers are reused in a pool with this maximum size.
+        # Each buffer has the size of 'maximum-frame-size'.
+        buffer-pool-size = 128
```
Comment (drewhk) [inline akka-remote/src/main/resources/reference.conf]: mention that it also works with a lower pool-size than used buffers, however, in that case there is a higher churn of DirectByteBuffers

**Review by drewhk — APPROVED** [review]

File: akka-remote/src/main/resources/reference.conf (hunk 2 above)
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: ok, will add

Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: added this:
```
        # This is not a hard upper limit on number of created buffers. Additional
        # buffers will be created if needed, e.g. when using many outbound
        # associations at the same time. Such additional buffers will be garbage 
        # collected, which is not as efficient as reusing buffers in the pool.
```

**Comment by patriknw** [comment]
rebased, will merge after validation

**Pull request merged**